app/src/main/java/com/afwsamples/testdpc/common/keyvaluepair/KeyValueBundleArrayFragment.java

```diff
@@ -40,43 +41,50 @@
 import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment
         .ARG_INITIAL_VALUE;
 import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment.ARG_KEY;
+import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment
+        .ARG_RESTRICTION_ENTRY;
 import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment.DialogType;
+import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment.RESULT_ENTRY;
 import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment.RESULT_KEY;
 import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment.RESULT_TYPE;
 import static com.afwsamples.testdpc.common.keyvaluepair.KeyValuePairDialogFragment.RESULT_VALUE;
 
 public class KeyValueBundleArrayFragment extends ManageAppFragment implements
-        OnEditButtonClickListener<Bundle>, OnDeleteButtonClickListener<Bundle> {
+        OnEditButtonClickListener<Integer>, OnDeleteButtonClickListener<Integer> {
     /**
      * Key of the editing bundle array.
      */
     private String mKey;
     /**
      * The current bundle list.
      */
-    private List<Bundle> mBundleList;
+    private ArrayList<Object> mBundleList;
```

Comment:
> Also, passing same data twice (as Bundle list and as RestrictionEnty list) looks strange.

I see it in another way. The Bundle object represents the data, while RestrictionEntry represents the schema of the data. So, we are not passing the same thing twice, but passing both the data schema and the data itself. The UI should enforce the schema, and do not allow user to violate it (e.g: put
a string to a int field). After users finish updating the data (bundle), we save the bundle straight away without conversion. In the ideal world, trust agent config is just a special case that the schema have no restriction on the data at all (empty list of RestrictionEntry), same as the case that app have no manifest restriction declared.

Thanks